src/framework/components/rigid-body/component.js

```diff
@@ -604,28 +618,19 @@ Object.assign(pc, function () {
          * in order to update the rigid body to match the Entity.
          */
         syncEntityToBody: function () {
-            var body = this.body;
+            var body = this.data.body;
             if (body) {
-                var pos = this.entity.getPosition();
-                var rot = this.entity.getRotation();
-
-                var transform = body.getWorldTransform();
-                var origin = transform.getOrigin();
-                origin.setValue(pos.x, pos.y, pos.z);
+                var wtm = this.entity.getWorldTransform();
+                ammoTransform.setFromOpenGLMatrix(wtm.data);
 
-                ammoQuat.setValue(rot.x, rot.y, rot.z, rot.w);
-                transform.setRotation(ammoQuat);
+                body.setWorldTransform(ammoTransform);
```

Comment:
Wondering what the difference is between body.setWorldTransform() and body.motionState.setWorldTransform()?

Why does this sync function sync both, but _updateKinematic only sets body.motionState?

Reply:
Exactly. I _think_ it's unnecessary. But right now, I'm trying not to break anything. 😄 This function could be modified further...

Reply:
For kinematics, yeah, you just need to set the motion state. I'm trying to figure out what needs to be done for all body types.

Reply:
I just read the [Bullet manual](https://github.com/bulletphysics/bullet3/blob/master/docs/Bullet_User_Manual.pdf) - see page 20. It says:

```
If you attempt to ask a body for its position through btCollisionObject::getWorldTransform
or btRigidBody::getCenterOfMassTransform, it will return the position at the end of the last
physics tick. That's useful for many things, but for rendering you will want some interpolation. Bullet
interpolates the transform of the body before passing the value to setWorldTransform.
If you want the non-interpolated position of a body [which will be the position as it was calculated at
the end of the last physics tick], use btRigidBody::getWorldTransform() and query the body directly.
```